### ISSUE TITLE
Grep shows service is running but I am not getting any numbers in the parameters

### ISSUE THREAD

**Issue by robfoo2015** [o-snoopy-o/dbus-canbus-battery#2]
Hi o-snoopy-o,  thanks very much for the work you have put into this project. I am new to the battery and inverter world and it's been a steep learning curve for me. But thanks in advance for any help you can give.

I had a similar issue to "ThanhMelb" and removed previous installs of dbuss-mqq, updated python to 3.8.18. I managed to get a result which shows the service running:

root@nanopi:/var/log# ps | grep dbus-canbus-battery.py
  719 root     31204 S    python3 /opt/victronenergy/dbus-canbus-battery/dbus-canbus-battery.py
 2595 root      2692 S    grep dbus-canbus-battery.py

However, the remote console just shows the battery with 0,0,0 for thedevice list and parameters. I have made up a cable which I will swap the three connections around and hope that works, but I suspect not as the parameters should still show some numbers.
The status on my victron multiplus ii-GX shows 0,0,0 for votage,watts,%.

Regards,
Robert

**Comment by ThanhMelb** [comment]
Hi Robert,

I recommend checking the cable again using a multimeter. Ensure that the end with PINs 1 and 2 is connected to the Samsung CAN port, while the end with PINs 7 and 8 is connected to the GX. Don’t forget to insert the terminator into the GX as well. Additionally, check the switch on the Samsung battery to ensure it is in the correct position. I realized I made a mistake by connecting to the wrong end previously.

You can also downgrade the GX firmware to version 3.52, reboot the GX, and then upgrade it to version 3.6. After that, follow the process to install the dbus-canbus-battery library. These steps should help resolve some issues with the library.

Good Luck 
Thanh

**Comment by robfoo2015** [comment]
Thanks Thanh, I will give that a go this weekend. The wiring is something I have trouble getting right and yes there still seems something wrong with the configuration on the victron. I have another go this weekend.
Regards Robert.

**Comment by Yarno069** [comment]
Hi Robert,

Did you have any luck with your issue?
I'm having the same, I have checked my cable but still nothing.
Thanks
Jan

**Comment by o-snoopy-o** [comment]
> Hi Robert,
> 
> Did you have any luck with your issue? I'm having the same, I have checked my cable but still nothing. Thanks Jan

If a can dump shows no data then you need to check your wiring as suggested above. If you are getting data then you need to make sure the service is running. Remember that by running grep command you should see 2 processes listed, if it’s just 1 the process is not running and it’s only the grep process shown, see the guide for correct screenshot showing both processes.

**Comment by Yarno069** [comment]
> > Hi Robert,
> > Did you have any luck with your issue? I'm having the same, I have checked my cable but still nothing. Thanks Jan
> 
> If a can dump shows no data then you need to check your wiring as suggested above. If you are getting data then you need to make sure the service is running. Remember that by running grep command you should see 2 processes listed, if it’s just 1 the process is not running and it’s only the grep process shown, see the guide for correct screenshot showing both processes.

![Image](https://github.com/user-attachments/assets/799173a6-087a-4918-9b2c-572e609277c7)
I think this is correct?

**Comment by o-snoopy-o** [comment]
> > > Hi Robert,
> > > Did you have any luck with your issue? I'm having the same, I have checked my cable but still nothing. Thanks Jan
> > 
> > 
> > If a can dump shows no data then you need to check your wiring as suggested above. If you are getting data then you need to make sure the service is running. Remember that by running grep command you should see 2 processes listed, if it’s just 1 the process is not running and it’s only the grep process shown, see the guide for correct screenshot showing both processes.
> 
> ![Image](https://github.com/user-attachments/assets/799173a6-087a-4918-9b2c-572e609277c7) I think this is correct?

No, you need to type: “candump can0” or type: ”candump can1”

**Comment by Yarno069** [comment]
> > > > Hi Robert,
> > > > Did you have any luck with your issue? I'm having the same, I have checked my cable but still nothing. Thanks Jan
> > > 
> > > 
> > > If a can dump shows no data then you need to check your wiring as suggested above. If you are getting data then you need to make sure the service is running. Remember that by running grep command you should see 2 processes listed, if it’s just 1 the process is not running and it’s only the grep process shown, see the guide for correct screenshot showing both processes.
> > 
> > 
> > ![Image](https://github.com/user-attachments/assets/799173a6-087a-4918-9b2c-572e609277c7) I think this is correct?
> 
> No, you need to type: “candump can0” or type: ”candump can1”

This is from the other thread https://github.com/o-snoopy-o/dbus-canbus-battery/issues/1

![Image](https://github.com/user-attachments/assets/7b153045-87d0-44d6-abcb-13d6a5e8b964)

**Comment by robfoo2015** [comment]
Yes, I got it communicating, I’ll send the cabling pin outs later tonight. I can confirm the Samsung will work with the Multiplus-II/GX with some tweaks using dbus-serialbattery. I couldn’t get it to run straight from the script as downloaded so I had it modified by Grok. It’s not perfect but at least I am seeing the battery.Sent from my iPhoneOn 29 Apr 2025, at 4:00 pm, Yarno069 ***@***.***> wrote:﻿
Yarno069 left a comment (o-snoopy-o/dbus-canbus-battery#2)




Hi Robert,
Did you have any luck with your issue? I'm having the same, I have checked my cable but still nothing. Thanks Jan

If a can dump shows no data then you need to check your wiring as suggested above. If you are getting data then you need to make sure the service is running. Remember that by running grep command you should see 2 processes listed, if it’s just 1 the process is not running and it’s only the grep process shown, see the guide for correct screenshot showing both processes.

image.png (view on web) I think this is correct?

No, you need to type: “candump can0” or type: ”candump can1”

This is from the other thread #1
image.png (view on web)

**Comment by robfoo2015** [comment]
@Yarno069 My current setup is using VE.Can on the Victron Multiplus-II/GX, it has a builtin Cerbo and does not have all the ports as described in most of these posts. I see you do have comms which is good. I also use CAN-Bus BMS LV 500kbt/s as that is the only protocol that does not show errors when you check network status on the remote console. The mappings I have are not perfect and are a work in progress. However, I am seeing the Samsung BMS and the data is mostly correct.

I'm using Dbus-Serialbattery instead of dbus-canbus-battery,  Here are some mappings that are "working".

 self.can_mappings = {
            "system_voltage": (0x500, 0, "U16", 0.01),
            "system_current": (0x500, 2, "S16", 1),
            "system_soc": (0x500, 4, "U8", 1),
            "system_soh": (0x500, 5, "U8", 1),
            "battery_charge_voltage": (0x502, 0, "U16", 0.1),
            "charge_current_limit": (0x502, 2, "U16", 0.1),
            "discharge_current_limit": (0x502, 4, "U16", 0.1),
            "battery_discharge_voltage": (0x502, 6, "U16", 0.1),
            "avg_cell_voltage": (0x503, 0, "U16", 0.001),
            "max_cell_voltage": (0x503, 2, "U16", 0.001),
            "min_cell_voltage": (0x503, 4, "U16", 0.001),
            "avg_cell_temperature": (0x504, 4, "S8", 1),
            "max_cell_temperature": (0x504, 5, "S8", 1),
            "min_cell_temperature": (0x504, 6, "S8", 1),
        }
